src/core/tsdf_submap_collection.cpp

```diff
@@ -41,6 +41,35 @@ void TsdfSubmapCollection::createNewSubMap(const Transformation &T_M_S,
   TsdfSubmap::Ptr tsdf_sub_map(
       new TsdfSubmap(T_M_S, submap_id, tsdf_map_config_));
   tsdf_sub_maps_.push_back(tsdf_sub_map);
+  id_to_submap_.emplace(submap_id, tsdf_sub_map);
+}
+
+void TsdfSubmapCollection::createNewSubMap(const Transformation &T_M_S) {
+  // Increment ID and create new submap
+  SubmapID new_ID = 0;
+  if (!tsdf_sub_maps_.empty()) new_ID = tsdf_sub_maps_.back()->getID() + 1;
```

Comment:
In general avoid single line if statements unless they are exceedingly simple. I would use braces here as it improves readability.